File: 2253/2253.py
class Solution:
    def dynamic_unpivoting_of_a_table(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.dynamic_unpivoting_of_a_table(root.left), self.dynamic_unpivoting_of_a_table(root.right))

File: 2253/2253.go
func dynamicUnpivotingOfATable(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := dynamicUnpivotingOfATable(root.Left)
	r := dynamicUnpivotingOfATable(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}